### ISSUE TITLE
Improve system-npm map/paths warnings

### ISSUE THREAD

**Issue by whitecolor** [stealjs/steal#544]
There are many warning while page loading like:
```
WARN: Could not find $css in node_modules. Ignoring.
```
Fctually for every module that mentioned in in `package.json` but not installed as dependency in `node_modules`.

Just wanted to make sure that you keep this issue in mind, didn't find it here.

**Comment by matthewp** [comment]
I dislike these as well. I think we should put them behind a higher log level, they are very rarely useful.

**Comment by whitecolor** [comment]
In what case these warning can be useful? For me the look like useless and misleading, and may be a annoying for new users.

**Comment by matthewp** [comment]
They would be useful if you were configuring a package and made a typo.

```json
{
  "map": {
    "cann/util/util": "can/util/jquery/jquery"
  }
}
```

```
WARN: Could not find cann in node_modules. Ignoring.
```

**Comment by whitecolor** [comment]
I think user will get another errror with wrong configuration while loading, and currently this warning react on everyting that is mapped in `map` or `paths", think it is not needed.

**Comment by marshallswain** [comment]
Maybe we can stick them behind a `{debug: true}` setting in the config, so they don't show up unless you want to troubleshoot what's going wrong.  What do you think?

**Comment by justinbmeyer** [comment]
They should show up by default and have people debug:false if they want them to disappear. I've found these useful.

**Comment by matthewp** [comment]
There's a couple of reasons why people complain about these warnings (I hear it constantly).

First, they do not know what they mean. The message

```
WARN: Could not find $css in node_modules. Ignoring.
```

Doesn't make sense to most people, they do not know what it is referring to and they do not know how to get rid of it. So the message could be clearer. Perhaps:

```
WARN: The map for $css is not an npm dependency, if this is a local module use ./$css instead.
```

Or something to that effect. This tells the user that we expect $css to be an npm module and how to fix their config to get rid of the warning.

**Comment by whitecolor** [comment]
Ah, so `./$css` here works actually the same way as `$css` for non node_modules. That actually solve the problem. And yes your variant of warning is really nice. Thanks.

**Comment by marshallswain** [comment]
:+1: Now it makes sense.

**Comment by matthewp** [comment]
Setting this as the 1.0.0 milestone so I make sure to do it by then, but will release sooner if anyone wants to take it.

**Comment by pYr0x** [comment]
@matthewp only rewrite the message? or implement a  `debug=true` config?

**Comment by matthewp** [comment]
Let's just update the message for now.

**Comment by pYr0x** [comment]
this is fixed in major branch of steal/system-npm
https://github.com/stealjs/system-npm/commit/e6e49f3b7e24c16006fa99bd94ed6f8578cdcf6a

**Comment by pYr0x** [comment]
this commit in system-npm is not in master and not published yet

**Comment by matthewp** [comment]
ill release it tomorrow.

**Comment by matthewp** [comment]
Closed by #759
